src/v/kafka/server/group.cc

```diff
@@ -1314,8 +1314,13 @@ void group::complete_offset_commit(
         // save the tp commit if it hasn't yet been seen, or we are completing
         // for an instance that is newer based on log offset
         auto o_it = _offsets.find(tp);
-        if (o_it == _offsets.end() || o_it->second.log_offset < md.log_offset) {
-            _offsets[tp] = md;
+        if (
+          o_it == _offsets.end()
+          || o_it->second.metadata.log_offset < md.log_offset) {
+            if (o_it != _offsets.end()) {
+                _offsets.erase(o_it);
+            }
+            _offsets.try_emplace(tp, std::move(md), _id, tp);
```

Comment:
When the offset is already in the map, I think you should update the metadata in place rather than removing and re-inserting. The former should be more efficient. Or am I misunderstanding what is happening here?